File: src/main/java/Airlines/AirlineTicket/Admin/Models/AdminService.java
package Airlines.AirlineTicket.Admin.Models;
import Airlines.AirlineTicket.Admin.Admin;

import java.util.List;
public interface AdminService {


        public List<Admin> getAllAdmins();


        public Admin getAdminById(Long id);


        public Admin saveAdmin(Admin admin);

        public void deleteAdmin(Long id);

}

File: frontend/src/services/AdminService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/admin';

class AdminService {
    getAllAdmins() {
        return axios.get(API_URL);
    }

    getAdminById(id) {
        return axios.get(`${API_URL}/${id}`);
    }

    createAdmin(admin) {
        return axios.post(API_URL, admin);
    }

    updateAdmin(id, admin) {
        return axios.put(`${API_URL}/${id}`, admin);
    }

    deleteAdmin(id) {
        return axios.delete(`${API_URL}/${id}`);
    }
}

export default new AdminService();
